mlflow/lightgbm/utils.py

```diff
@@ -0,0 +1,110 @@
+import importlib
+import json
+import os.path
+import warnings
+import numpy as np
+
+
+def _label_encoder_to_json(le):
+    """Returns a JSON compatible dictionary"""
+    meta = {}
+    for k, v in le.__dict__.items():
+        if isinstance(v, np.ndarray):
+            meta[k] = v.tolist()
+        else:
+            meta[k] = v
+    return meta
+
+
+def _label_encoder_from_json(doc):
+    """Load the encoder back from a JSON compatible dict"""
+    from lightgbm.compat import _LGBMLabelEncoder
+
+    le = _LGBMLabelEncoder()
+    meta = {}
+    for k, v in doc.items():
+        if k == "classes_":
+            le.classes_ = np.array(v) if v is not None else None
+            continue
+        meta[k] = v
+    le.__dict__.update(meta)
+    return le
+
+
+def _save_lgb_attr(model_dir, fname, attr_dict):
+    with open(os.path.join(model_dir, "{}.json".format(fname)), "w") as f:
+        json.dump(attr_dict, f)
+
+
+def _load_lgb_attr(model_dir, fname):
+    try:
+        with open(os.path.join(model_dir, "{}.json".format(fname))) as f:
+            attr = json.load(f)
+        return attr
+    except IOError:
+        return None
+
+
+def _save_lgb_model(lgb_model, model_path) -> None:
```

Comment:
Following from https://github.com/mlflow/mlflow/pull/5130#issuecomment-992784713, I think it's probably safest to use cloudpickle to save / load the model, since the LightGBM developers could make breaking changes that break the serialization / deserialization code. Thank you so much for taking the time to reach out to the LightGBM community and get some insight into the recommended best practices here.